### PR TITLE
Ana/fix transaction item in extension

### PR DESCRIPTION
**Pull request by Bitcoinera**
Closes #ISSUE

**Description:**

Uploaded to try in the extension

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

File: src/components/transactions/TransactionItem.vue [hunk 1]
```diff
@@ -73,7 +74,9 @@ export default {
     }
   },
   data: () => ({
-    show: false
+    show: false,
+    showValidators: false,
```
Comment (jbibla) [inline src/components/transactions/TransactionItem.vue]: why do we need this boolean?

File: src/components/transactions/TransactionItem.vue [hunk 2]
```diff
@@ -73,7 +74,9 @@ export default {
     }
   },
   data: () => ({
-    show: false
+    show: false,
```
Comment (jbibla) [inline src/components/transactions/TransactionItem.vue]: do we still need this is `show-validators`?

File: config.js [hunk 3]
```diff
@@ -16,6 +16,7 @@ export default {
   // Ledger
   CosmosAppTestModeAllowed: false,
   mobileApp: Boolean(process.env.MOBILE_APP),
+  isExtension: Boolean(process.env.EXTENSION),
```
Comment (jbibla) [inline config.js]: 👍

**Comment by jbibla** [comment]
@Bitcoinera cha cha cha... changelog 💃

**Comment by Bitcoinera** [comment]
> i like the `isExtension` flag. however, this feels like a lot of extra logic for something that should just work as expected (validators should display in transactions). maybe we should just wait for transaction abstraction? is that the issue here?

Well, I couldn't find any other solution apart from adding this variable and the issue doesn't lie there really... Since this is actually not solving the validators not displaying issue but the empty box issue (tx metada).

If you click on the TransactionTiem the Metadata slide pops out. How can you fix it otherwise but telling the slide not to pop when it is in Extension?

The validators not displaying issue is a different one and have it almost fixed it. I am just getting the pic from the DB. There is no other alternative (transaction abstraction doesn't help either here. This is just how the extension works)

File: src/components/transactions/TransactionItem.vue (hunk 1 above)
Comment (Bitcoinera) [reply src/components/transactions/TransactionItem.vue]: this is to make the show-all-validators thing in the WithdrawRewardsMessageDetails component independent from the `show` boolean coming from TransactionItem (so in the extension you actually get the cool effect of validators going vertical with their names)

**Comment by jbibla** [comment]
ok i think i understand. thanks for the clarification.

> If you click on the TransactionItem the Metadata slide pops out. How can you fix it otherwise but telling the slide not to pop when it is in Extension?

why can't we display the metadata, like we do in the FE? why does the tx in the extension need to be different?

**Comment by Bitcoinera** [comment]
> why can't we display the metadata, like we do in the FE? why does the tx in the extension need to be different?

The problem is that metadata doesn't exist yet at that point since you are about to sign for that transaction to be broadcast to the network

File: src/components/transactions/TransactionItem.vue (hunk 2 above)
Comment (Bitcoinera) [reply src/components/transactions/TransactionItem.vue]: no, this one is the `show` for the transaction metadata, so yes, we still need it here

Comment (jbibla) [reply src/components/transactions/TransactionItem.vue]: so this should be called `showMetaData` probably 😸

**Comment by jbibla** [comment]
> The problem is that metadata doesn't exist yet at that point since you are about to sign for that transaction to be broadcast to the network

ah! brilliant! thank you! i did not understand the problem.

you are absolutely correct. this is great.

File: src/components/transactions/TransactionItem.vue (hunk 2 above)
Comment (jbibla) [reply src/components/transactions/TransactionItem.vue]: wait, there is already a prop called `showMetaData` on the `TransactionMetaData` component.

File: src/components/transactions/TransactionItem.vue [hunk 4]
```diff
@@ -106,9 +109,14 @@ export default {
...
   },
   methods: {
+    /* istanbul ignore next */
     toggleDetail(event) {
-      if (event.target.className !== `address`) {
+      if (this.isExtension) {
+        this.showValidators = !this.showValidators
+        return
+      } else if (event.target.className !== `address`) {
         this.show = !this.show
```
Comment (jbibla) [inline src/components/transactions/TransactionItem.vue]: this isn't being used anymore, right?

Comment (Bitcoinera) [reply src/components/transactions/TransactionItem.vue]: No, you are right. I just deleted it

**Comment by faboweb** [comment]
horribly complex. why not just add a `v-if="!isExtension"` to line 17?

**Comment by Bitcoinera** [comment]
> horribly complex

xD

**Comment by Bitcoinera** [comment]
Closed on favor of #3590

**Pull request closed without merging**